internal/services/containerapps/helpers/container_app_environment.go

```diff
@@ -8,15 +8,42 @@ import (
 
 	"github.com/hashicorp/go-azure-helpers/lang/pointer"
 	"github.com/hashicorp/go-azure-sdk/resource-manager/containerapps/2023-05-01/managedenvironments"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/features"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
 )
 
-const consumption = "Consumption"
+type WorkloadProfileSku string
+
+const (
+	WorkloadProfileSkuConsumption WorkloadProfileSku = "Consumption"
+	WorkloadProfileSkuD4          WorkloadProfileSku = "D4"
+	WorkloadProfileSkuD8          WorkloadProfileSku = "D8"
+	WorkloadProfileSkuD16         WorkloadProfileSku = "D16"
+	WorkloadProfileSkuD32         WorkloadProfileSku = "D32"
+	WorkloadProfileSkuE4          WorkloadProfileSku = "E4"
+	WorkloadProfileSkuE8          WorkloadProfileSku = "E8"
+	WorkloadProfileSkuE16         WorkloadProfileSku = "E16"
+	WorkloadProfileSkuE32         WorkloadProfileSku = "E32"
+)
```

Comment:
I was surprised this isn't documented in the Swagger, but checking the SDK we're not generating it - could we add a NOTE for this one

```suggestion
// NOTE: the Workload Profile SKUs aren't defined in the Swagger definition so we define them here

const (
	WorkloadProfileSkuConsumption WorkloadProfileSku = "Consumption"
	WorkloadProfileSkuD4          WorkloadProfileSku = "D4"
	WorkloadProfileSkuD8          WorkloadProfileSku = "D8"
	WorkloadProfileSkuD16         WorkloadProfileSku = "D16"
	WorkloadProfileSkuD32         WorkloadProfileSku = "D32"
	WorkloadProfileSkuE4          WorkloadProfileSku = "E4"
	WorkloadProfileSkuE8          WorkloadProfileSku = "E8"
	WorkloadProfileSkuE16         WorkloadProfileSku = "E16"
	WorkloadProfileSkuE32         WorkloadProfileSku = "E32"
)
```